File: src/learning_communication/scripts/listener.py
#! /usr/bin/python
import rospy
from std_msgs.msg import String


def myCallback(msg):
    rospy.loginfo("receive " + msg.data)

def listener():
    lsner = rospy.Subscriber("talker", String, queue_size=100, callback=myCallback)
    rospy.init_node("listener", anonymous=True)
    rospy.spin()

if __name__ == "__main__":
    try:
        listener()
    except rospy.ROSInterruptException:
        pass

File: src/learning_communication/src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void myCallback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("msg is : %s", msg->data.c_str());
}


int main(int argc, char *argv[])
{
    ros::init(argc, argv, "listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("talker", 1000, myCallback);
    ros::spin();
    return 0;
}
